### PR TITLE
Enable validation on default webhook

### PR DESCRIPTION
**Pull request by dbadura**
This PR add:
- validator service which can be used by admission and operator
- implement validation logic in default webhook

### CODE REVIEW

File: internal/admission/defaulting.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	w.logger.Infof("pod was validated: %s, %s", pod.ObjectMeta.GetName(), pod.ObjectMeta.GetNamespace())
+	return admission.PatchResponseFromRaw(req.Object.Raw, fBytes)
+}
+
+func (w *DefaultingWebHook) InjectDecoder(decoder *admission.Decoder) error {
+	w.decoder = decoder
+	return nil
+}
+
+func (w DefaultingWebHook) checkPod(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, error) {
```
Comment (m00g3n) [inline internal/admission/defaulting.go:69]: can we change the name of this method - it does more than just checking pod

File: internal/admission/defaulting.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	if labelToApply == "" {
+		return nil, nil
+	}
+
+	labeledPod := pod.DeepCopy()
+	labeledPod.ObjectMeta.Labels = w.labelPod(labeledPod.ObjectMeta.Labels, pkg.PodValidationLabel, labelToApply)
+	return labeledPod, nil
+}
+
+func (w *DefaultingWebHook) labelPod(labels map[string]string, labelKey, labelValue string) map[string]string {
```
Comment (m00g3n) [inline internal/admission/defaulting.go:89]: - we don't need to have this as method receiver
- this method it is used only in one place - inside `checkPod`
- we could merge both methods and have no method receiver and a meaningful name for `checkPod`

File: internal/validate/pod.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	})
+
+	return admitResult, nil
+}
+
+func (a *podValidator) IsValidationEnabledForNS(ns *corev1.Namespace) bool {
+	return ns.GetLabels()[pkg.NamespaceValidationLabel] == pkg.NamespaceValidationEnabled
+}
+
+func (a *podValidator) admitPodImage(image string) (ValidationResult, error) {
```
Comment (m00g3n) [inline internal/validate/pod.go:75]: shouldn't it be `validateImage` instead of `admitPodImage`?

File: internal/validate/pod.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+func NewPodValidator(imageValidator ImageValidatorService) PodValidator {
+	return &podValidator{
+		imageValidator,
+	}
+}
+
+func (a *podValidator) ValidatePod(ctx context.Context, pod *corev1.Pod, ns *corev1.Namespace) (ValidationResult, error) {
+	l := log.FromContext(ctx)
```
Comment (m00g3n) [inline internal/validate/pod.go:43]: If there is no plan for `ctx` we could remove it.

Comment (dbadura) [reply internal/validate/pod.go:43]: There is a plan for `ctx`, so I will keep it.

File: internal/admission/defaulting.go (hunk 2 above)
Comment (dbadura) [reply internal/admission/defaulting.go:89]: I reorganized the code

File: .gitignore [hunk 5]
```diff
@@ -23,3 +23,6 @@ vendor
 *.swo
 *~
 *.DS_Store
+
+operator
```
Comment (dbadura) [inline .gitignore:27]: This bug will be fixed in: https://github.com/kyma-project/warden/pull/25

**Review by m00g3n — APPROVED** [review]
LGTM

**Pull request merged**